rts/Game/PreGame.cpp

```diff
@@ -168,37 +168,58 @@ int CPreGame::KeyPressed(int keyCode, int scanCode, bool isRepeat)
 	return 0;
 }
 
+void CPreGame::AsyncExecute(CPreGame::AsyncExecFuncType execFunc, const std::string& argument)
+{
+	pendingTask = std::async(std::launch::async,
+		[execFunc, this](const std::string& argument) {
+			return std::invoke(execFunc, this, argument);
+		},
+		argument /*will make a copy for thread safety*/
```

Comment:
Will it actually? This passes a reference and the lambda in the line above also expects a reference, where does the copy materialize?

Reply:
`std::async` and alike actually store the arguments, if I wanted to pass argument as a reference I'd need to use std::cref/std::ref (and get crashes).
But I think your note makes sense it's best to copy explicitly.

Reply:
I think a comment is enough, I just didnt trust the current one too much since it just looks copy-pasted a bit. How about
```suggestion
		argument /* std::async always makes a copy, even if you pass a reference */
```

Reply:
The copy via lambda capture works as well tho.

Reply:
Too late, the new code will remove ambiguity.